core/startos/src/service/transition/restore.rs

```diff
@@ -73,9 +68,9 @@ impl Handler<Restore> for ServiceActor {
         if let Some(t) = old {
             t.abort().await;
         }
-        match transition.await {
-            None => Err(Error::new(eyre!("Restoring canceled"), ErrorKind::Unknown)),
-            Some(x) => Ok(x),
+        match recv_res.await {
```

Comment:
can you explain the difference between using a transition vs oneshot here?

Reply:
the transition was just a handle to the completion of the future but did not actually return the success or failure of it